### PR TITLE
Brings in NuttX Multi-PHY runtime PHY detection and uses it one px4_fmu-v6xrt

### PR DESCRIPTION
**Pull request by davids5**
There are several features added for the px4_fmu-v6xrt running on the NXP T1 1170 Base.

### CODE REVIEW

File: boards/px4/fmu-v6xrt/src/spi.cpp [hunk 1]
```diff
@@ -41,47 +41,65 @@
 #include <drivers/drv_sensor.h>
 #include <nuttx/spi/spi.h>
 
-#include <px4_platform_common/px4_config.h>
-#include <px4_log.h>
+constexpr px4_spi_bus_all_hw_t px4_spi_buses_all_hw[BOARD_NUM_SPI_CFG_HW_VERSIONS] = {
+	initSPIHWVersion(V6XRT_00, {
+		initSPIBus(SPI::Bus::LPSPI1, {
+			initSPIDevice(DRV_IMU_DEVTYPE_ICM42688P, SPI::CS{GPIO::Port2, GPIO::Pin11}, SPI::DRDY{GPIO::Port3, GPIO::Pin19}), /* GPIO_EMC_B2_01 GPIO2_IO11, GPIO_AD_20, GPIO3_IO19 */
```
Comment (PetervdPerk-NXP) [inline boards/px4/fmu-v6xrt/src/spi.cpp:47]: SPI1 is a icm42686p in the production batch

Comment (davids5) [reply boards/px4/fmu-v6xrt/src/spi.cpp:47]: @PetervdPerk-NXP Good catch

**Comment by davids5** [comment]
@dagar This is read to come in. 
Repointed to PX4 NuttX head and rebased on current main.

**Review by PetervdPerk-NXP — APPROVED** [review]

**Comment by davids5** [comment]
@dagar Since [this](https://github.com/PX4/PX4-Autopilot/pull/22608/commits/11d3d969e1993fdabeb9aa091eb856caba961604) touches the RCInput I would like you to have a look.

**Review by dagar — APPROVED** [review]

**Pull request merged**